modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -61,12 +72,32 @@
     public Node getNode();
 
     /**
-     * Called by a Skinnable when the Skin is replaced on the Skinnable. This method
-     * allows a Skin to implement any logic necessary to clean up itself after
-     * the Skin is no longer needed. It may be used to release native resources.
+     * Called once when {@link Skin} is set, after the
+     * previous skin, if any, has been uninstalled via its {@link #dispose()} method.
+     * This method allows a Skin to register listeners, add child nodes, set
+     * required properties and/or event handlers.
+     * <p>
+     * Application code must not call this method.
+     * <p>
+     * The default implementation of this method does nothing.
+     *
+     * @implNote
+     * Most implementations of Skin
+     * do not need to implement {@code install} unless they must set one or more
+     * properties in the corresponding Skinnable.
```

Comment:
I think this might be better:

```suggestion
     * Skins only need to implement {@code install} if they need to make direct changes to the control
     * like overwriting properties or event handlers. Such skins should ensure these changes are undone in
     * their {@link #dipose()} method.
```

Reply:
Your version is so much better, than you.